### PR TITLE
Merging Fixes for Xamarin Android SDK for: (1) Compilation Issues. (2) Re-connect if internet connection is up again.

### PR DESCRIPTION
**Pull request by iitr-ankur**
The changes are actually very small, but the diff is not working correctly and showing same text also being deleted and added.

### CODE REVIEW

File: mono-for-android/PubNubMessaging.Example/MainActivity.cs [hunk 1]
```diff
@@ -1,929 +1,929 @@
...
-
-            TextView txtViewLog = FindViewById<TextView> (Resource.Id.txtViewLog);
-            txtViewLog.Text = "";
-            try {
-                this.m_Drawer = this.FindViewById<DrawerLayout> (Resource.Id.drawer_layout);
-                this.m_DrawerList = this.FindViewById<ListView> (Resource.Id.left_drawer);
-                this.m_DrawerList.Adapter = new ArrayAdapter<string> (this, Resource.Layout.ItemMenu, Sections);
-                this.m_DrawerList.ItemClick += (sender, args) => ListItemClicked (args.Position);
-
-                this.m_Drawer.SetDrawerShadow (Resource.Drawable.drawer_shadow_dark, (int)GravityFlags.Start);
```
Comment (iitr-ankur) [inline mono-for-android/PubNubMessaging.Example/MainActivity.cs]: Removed the cast to int. This is the only change in this file.

File: mono-for-android/PubNubMessaging/ClientNetworkStatus.cs [hunk 2]
```diff
@@ -1,297 +1,297 @@
...
+                return _machineSuspendMode;
+            }
+            set {
+                _machineSuspendMode = value;
+            }
+        }
+        /*#if(__MonoCS__ && !UNITY_IOS && !UNITY_ANDROID)
+        static UdpClient udp;
+        #endif*/
+        #if(UNITY_IOS || UNITY_ANDROID || MONODROID || __MonoCS__)
```
Comment (iitr-ankur) [inline mono-for-android/PubNubMessaging/ClientNetworkStatus.cs]: Only Change here of Adding missing Monodroid .

**Comment by crimsonred** [comment]
@iitr-ankur , can you explain what compilation issues were you facing?

Reconnect also works well with the existing code. What's the scenario where it was failing. 

Which IDE are you using and whats its version?

**Comment by iitr-ankur** [comment]
Compilation issues: 
Solution: c-sharp\mono-for-android\Pubnub-messaging solution file.
IDEs: Xamarin Studio and Visual Studio. 
Set PubNubMessaging.Example project as start-up project. It was not compiling and giving multiple compilation errors even after adding "Monodroid" compilation symbol. Also,  in MainActivity.cs: 
SetDrawerShadow (Resource.Drawable.drawer_shadow_dark, (int)GravityFlags.Start);
The above function expects the enum and not int as the second argument. 

You can verify these by just building the solution and defining Monodroid as the compilation symbol.


Re-connection issue:

After fixing the compilation issues, run and launch the app in either emualtor(genymotion) or real device. (motorola-2nd gen). The publish, subscribe etc. works. Then switch mobile to Airplane mode or disconnect wifi. You will start seeing messages that it is trying to connect to internet and 
"Detected internet connection problem. Retrying connection attempt 5 of 50". Then re-enable wifi. Open browser in mobile to verify internet connection is back. Also, wait for some time to continue to see these "Retrying.." messages with increasing connection attempt. But it never gets connected back. Looks like the code to reconnect is not actually trying a reconnect, it is just logging reconnect attempts without actually calling the same.  After my fix, it is able to connect back within 1-2 more retries. 

Please try it at your end once. I had to spend many days in understanding the code, and making changes to test and verify this. The changes are small though.

**Comment by crimsonred** [comment]
@iitr-ankur , looks like you are targeting .net framework using VS, you need to define both "__MonoCS__" and "MONODROID" in the compilation symbols to make it work using VS. The ClientNetworkStatus.cs uses the mono specific code in this case.

Whats the version of the IDE's (for the SetDrawerShadow issue)

**Comment by iitr-ankur** [comment]
IDEs:
Xamarin Studio 5.7.1

Microsoft Visual Studio Ultimate 2013
Version 12.0.31101.00 Update 4
Microsoft .NET Framework
Version 4.5.51650

The documentation doesn't mention defining MonoCS also: 
https://github.com/pubnub/c-sharp/tree/master/mono-for-android

Also, what about the re-connect issue?

**Comment by crimsonred** [comment]
@iitr-ankur, defining `__MonoCS__` will fix the reconnect issue.

**Comment by iitr-ankur** [comment]
@crimsonred 

Thanks for your help. Defining __MonoCS__ fixed the issues except for the SetDrawerShadow issue.
Please update the documentation @ ( https://github.com/pubnub/c-sharp/tree/master/mono-for-android ) also which misled me and wasted so much time.

**Pull request closed without merging**